### PR TITLE
fix(fcm): handle missing notification types and add tests

### PR DESCRIPTION
**Pull request by othmanehousni**
This PR completes FCM notification handling by adding support for missing notification types and aligning the test suite accordingly.

### What

* Added full FCM processing support for:

  * **Event invitations**
  * **Organization member invitations**
* Extended `FCMNotificationHandler` to correctly parse payload data and create the corresponding notification models.
* Added comprehensive unit tests covering:

  * Correct notification creation for each new type
  * Graceful handling of missing or optional payload fields
  * Default fallback values where appropriate

### Why

Some notification types were already defined in the domain model but were not handled when received via Firebase Cloud Messaging. This caused valid push notifications to be silently ignored and never persisted. Completing the FCM handler ensures all supported notification types are reliably delivered and stored.

### How

* Routed new `NotificationType` values in `processMessage`
* Implemented dedicated handlers for event and organization invitations
* Ensured safe parsing with early exits on missing mandatory fields
* Added exhaustive unit tests to validate both success paths and edge cases

### Testing

* Added unit tests for all new handlers
* Existing tests adapted where needed
* All tests passing locally

Closes: #385

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/service/FCMNotificationHandler.kt [hunk 1]
```diff
@@ -91,6 +94,66 @@ class FCMNotificationHandler(
...
   }
 
+  /**
+   * Processes an event invitation notification
+   *
+   * @param data The notification data
+   * @param userId The user ID
+   */
+  fun processEventInvitation(data: Map<String, String>, userId: String) {
+    val eventId = data["eventId"] ?: return
```
[the remark this answers is not in the document]
Comment (othmanehousni) [reply app/src/main/java/com/github/se/studentconnect/service/FCMNotificationHandler.kt]: added logging 💯

File: app/src/main/java/com/github/se/studentconnect/service/FCMNotificationHandler.kt [hunk 2]
```diff
@@ -91,6 +94,66 @@ class FCMNotificationHandler(
...
+    val eventTitle = data["eventTitle"] ?: "Event"
```
[the remark this answers is not in the document]
Comment (othmanehousni) [reply app/src/main/java/com/github/se/studentconnect/service/FCMNotificationHandler.kt]: yuppp removed hardcoded strings
